python/mxnet/optimizer.py

```diff
@@ -299,8 +300,13 @@ def update(self, index, weight, grad, state):
         if state:
             mom = state
             mom[:] *= self.momentum
-            mom[:] += -lr * (grad + wd * weight)
-            weight[:] += mom
+            grad += wd * weight
```

Comment:
yes, but the problem is mixing them together makes it hard to figure out which part is which and hard to modify either of them. Also it makes it hard for new users to find nestrov because it's hidden in sgd